src/main/java/org/commonwl/view/cwl/CWLService.java

```diff
@@ -1056,4 +1060,26 @@ private Object extractRun(Map<String, Object> step) {
     }
     return null;
   }
+
+  private String normaliseLicenseLink(String licenseLink) {
+    if (licenseLink == null) {
+      return null;
+    }
+    if (licenseLink.startsWith(LicenseUtils.SPDX_PREFIX)) {
+      return licenseLink;
+    }
+    String httpsLicenseLink = StringUtils.stripEnd(licenseLink.replace("http://", "https://"), "/");
+    return switch (httpsLicenseLink) {
+      case "https://www.apache.org/licenses/LICENSE-2.0",
+          "https://opensource.org/licenses/Apache-2.0" -> LicenseUtils.SPDX_PREFIX
+          + SpdxLicense.APACHE_2_0.id;
+      case "https://mit-license.org", "https://opensource.org/licenses/MIT" -> LicenseUtils
+              .SPDX_PREFIX
+          + SpdxLicense.MIT;
+      case "https://www.gnu.org/licenses/agpl.txt",
+          "https://www.opensource.org/licenses/AGPL-3.0" -> LicenseUtils.SPDX_PREFIX
+          + SpdxLicense.AGPL_3_0;
+      default -> licenseLink;
```

Comment:
Good idea to move from the template to the backend.

Maybe in the future we could also load the spdx [JSON data](https://github.com/spdx/license-list-data/blob/main/json/licenses.json), quickly load it with `gson` or something similar, and implement methods like `SpdxCache.findByUrl` or something similar.

I think that way we could remove this switch and return the spdx prefixed URL for most licenses?

Reply:
This is a good idea. Here we should probably index by the `seeAlso` field, forcing the initial `https://` prefix for uniformity. Let me try to do it

Reply:
Ok I tried to do it by creating an application-scoped bean object